### PR TITLE
Block Library: Deprecate the Post Comment block

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Now that Comments Query Loop and Comment Template blocks are up and running we can finally deprecate the Post Comment block. The procedure is similar to what we did to the Text Columns block in the past:
- hide the block from the inserter
- update the title and description of the block to reflect the deprecation 

In this case, we might be able to remove the block completely after a few Gutenberg plugin releases because this block never went out of the experimental phase.

In addition to that, I update all blocks to use the Comment Template block as a template. I also changed the title, description, and category of several blocks to align with how Query Loop related blocks look like.

Bonus refactoring is based on the comment from @michalczaplinski in https://github.com/WordPress/gutenberg/pull/35965#discussion_r737734840 about the redundant `className` usage in `edit` method. It's handled with `useBlockProps` so we can remove it.

## How has this been tested?

1. Open the Site Editor for a block theme.
2. Navigate to the Single Post template.
3. Ensure you can't find the Post Comment block in the inserter. You should be able to insert it manually in the code editing mode when enabled.
4. Insert a Comments Query Loop block.
5. Insert more blocks inside the Comment Template block
    1. Comment Author Avatar.
    2. Comment Edit Link.
    3. Comment Reply Link. 
6. Ensure that the comments are rendered on the frontend after saving the template. 

## Screenshots 

All blocks in action inside the template:
https://user-images.githubusercontent.com/699132/139238006-d2d2610a-4023-4c5e-8019-e01dba733f36.mov

HTML inside the block editor (it contains all custom class names applied):
Screen Shot 2021-10-28 at 11 31 33

HTML on the frontend (it contains all custom class names applied):
Screen Shot 2021-10-28 at 11 31 45

Searching in the inserter inside the Comment Template block:
Screen Shot 2021-10-28 at 12 08 51

Searching in the inserter outside of the Comment Template block:
Screen Shot 2021-10-28 at 12 11 03

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/post-comment/block.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
 	"apiVersion": 2,
 	"name": "core/post-comment",
-	"title": "Post Comment",
+	"title": "Post Comment (deprecated)",
 	"category": "theme",
-	"description": "Post comment.",
+	"description": "This block is deprecated. Please use the Comment Template block instead.",
```
Comment (ntsekouras) [inline packages/block-library/src/post-comment/block.json]:
```suggestion
	"description": "This block is deprecated. Please use the Comments Query Loop block instead.",
```

**Review by ntsekouras — APPROVED** [review]
Thanks Grzegorz! Looks good! 👍

File: packages/block-library/src/post-comment/block.json (hunk 1 above)
Comment (gziolo) [reply packages/block-library/src/post-comment/block.json]: Right, makes sense 👍🏻

**Comment by noisysocks** [comment]
We're going to ship Post Comments in WP 5.9 as the Comments Query Loop isn't ready (see https://github.com/WordPress/gutenberg/pull/35979#issuecomment-956681949).

So please wait until after Gutenberg 11.9 / WordPress 5.9 before merging this.

**Comment by gziolo** [comment]
> We're going to ship Post Comments in WP 5.9 as the Comments Query Loop isn't ready (see [#35979 (comment)](https://github.com/WordPress/gutenberg/pull/35979#issuecomment-956681949)).
> 
> So please wait until after Gutenberg 11.9 / WordPress 5.9 before merging this.

It doesn't need to wait. There are two blocks: Post Comments and Post Comment. This PR deprecates only the Post Comment block that was never close to useful - you had to explicitly pass the comment id to display content. Post Comments doesn't use Post Comment internally.

**Comment by noisysocks** [comment]
Aha, my mistake 😄

**Pull request merged**